EIPS/eip-6733.md

```diff
@@ -0,0 +1,71 @@
+---
+eip: 6733
+title: Revert creation in case of collision
+description: Revert contract creation if address already has code
+author: Renan Rodrigues de Souza (@RenanSouza2)
+discussions-to: https://ethereum-magicians.org/t/eip-revert-on-address-collision/13442
+status: Draft
+type: Standards Track
+category: Core
+created: 2023-03-20
+---
+
+## Abstract
+
+There is no current definition to what happens when a contract creation happens in an address with code already deployed. This fix prevents an attack of deploying contract code and later changing the code arbitrarily.
+
+## Motivation
+
+In [EIP-3607](./eip-3607.md) it was estimated that to create an address collision it would take something about 10 billion USD and this number decreases as computers processing power grow.
```

Comment:
```suggestion
In [EIP-3607](./eip-3607.md) it was estimated to require 10 billion USD in computing power to create an address collision. This number decreases as computers become more efficient over time.
```
It's worth noting that this is an estimate for *any* address collision. To create an address collision at a specific address, it would require much more computing power, costing around $12,089,258,196,146,292,000,000,000,000,000,000.

The most likely effect of this EIP would be to increase the cost of using the Ethereum network, either as a validator, node, or transaction signer.

Reply:
I inclused those changes as well